signer/xpi/xpi.go

```diff
@@ -224,6 +267,22 @@ func (s *XPISigner) SignFile(input []byte, options interface{}) (signedFile sign
 		return nil, errors.Wrap(err, "xpi: error parsing cose_algorithms options")
 	}
 
+	if s.Mode == ModeAddOnWithRecommendation {
+		recFileBytes, err := s.makeRecommendationFile(opt, cn)
+		if err != nil {
+			return nil, errors.Wrap(err, "xpi: error making recommendation file from options")
+		}
+		input, err = appendFileToZIP(input, s.recommendationFilePath, recFileBytes)
+		if err != nil {
+			return nil, errors.Wrap(err, "xpi: error append recommendation file to XPI")
+		}
+	} else {
+		input, err = removeFileFromZIP(input, s.recommendationFilePath)
+		if err != nil {
+			return nil, errors.Wrap(err, "xpi: error removing recommendation file from XPI")
+		}
+	}
```

Comment:
perhaps `removeFileFromZIP` should always be invoked, because we want to remove existing recommendations anyway.
```go
	input, err = removeFileFromZIP(input, s.recommendationFilePath)
	if err != nil {
		return nil, errors.Wrap(err, "xpi: error removing recommendation file from XPI")
	}
	if s.Mode == ModeAddOnWithRecommendation {
		recFileBytes, err := s.makeRecommendationFile(opt, cn)
		if err != nil {
			return nil, errors.Wrap(err, "xpi: error making recommendation file from options")
		}
		input, err = appendFileToZIP(input, s.recommendationFilePath, recFileBytes)
		if err != nil {
			return nil, errors.Wrap(err, "xpi: error append recommendation file to XPI")
		}
	}
```